code/modules/mining/machine_processing.dm

```diff
@@ -5,6 +5,33 @@
 /obj/machinery/mineral
 	var/input_dir = NORTH
 	var/output_dir = SOUTH
+	var/turf/input_turf = null // The turf that the machines will be listening to for ores/items
+	var/needs_item_input = FALSE // whether or not the machine needs to pick up items, i.e. the ORM picking up ores

```

Comment:
I still think this would be better off in a component that takes a callback.

Reply:
Would this be included in the same component that would handle the I/O?